src/components/LHNOptionsList/OptionRowLHNData.js

```diff
@@ -186,6 +198,11 @@ export default React.memo(
                 key: ({fullReport}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${fullReport.parentReportID}`,
                 canEvict: false,
             },
+            // Ideally, we aim to access only the last transaction for the current report by listening to changes in reportActions.
+            // In some scenarios, a transaction might be created after reportActions have been modified.
+            // This can lead to situations where `lastTransaction` doesn't update and retains the previous value.
+            // However, performance overhead of this is minimized by using memos inside the component.
+            receiptTransactions: {key: ONYXKEYS.COLLECTION.TRANSACTION},
```

Comment:
I'm not sure that it was fully realized, but this means that _every_ option in the LHN is connected to the _entire_ transaction collection. That means that when _any_ transaction is updated, every single option in the LHN is re-rendered. That seems like it's bad for performance.

Reply:
bump @ygshbht and @luacmartins

Reply:
@tgolen Yes, that results in additional renders. However, I don't perceive it as a significant performance issue as transaction updates are infrequent and memoization is implemented within the component. But indeed, there's room for improvement. What specific optimizations do you propose?

Reply:
OK, I don't really fully understand that comment. Some of the questions I have are:

- "In some scenarios" what scenarios exactly?
- Where does `lastTransaction` come from and why doesn't it update?

It seems like if the root problem for that was fixed, then there wouldn't be a need for this workaround.

Reply:
>"In some scenarios" what scenarios exactly?

Just the normal/ any case.

By "In some scenarios" i mean the probabilty of that happening is like 3/10 for the same situations. Honestly, if i remember it might even be "Most of the time" (i.e 8/10 or 10/10) instead of "In some scenarios". 

> Where does lastTransaction come from and why doesn't it update?

`lastTranscation` comes from the last `reportAction ` whose status you wanna show in the LHN subtitle  

From this comment of mine in the code, I infer: 
> // In some scenarios, a transaction might be created after reportActions have been modified.

That `transaction ` is created after `reportActions` have been udpated and hence you don't have any way of detecting when transaction further updates as you don't have access to it. So you are not able to detect the changes and show it in LHN

Reply:
OK thanks! Bear with me as I try to figure out how all this works. It seems to me like if there is a `reportAction` that you're connected to, and it has a reference to a `transactionID`, then when the transaction is created, the `transactionID` of the `reportAction` would go from `null` to `<transactionID`. At that point, you can connect to the transaction (via `withOnyx`) and have access to it.

Or am I still missing something?

Reply:
My initial approach, instead of 
`receiptTransactions: { key: ONYXKEYS.COLLECTION.TRANSACTION }`

was

```
receiptTransaction: {
    key: ({ reportActions }) => {
        const lastReportAction = getLastReportAction(reportActions);
        const lastTransaction  = getTransaction(lastReportAction);
        const transactionId = getTransactionId(lastTransaction);
        return `ONYXKEYS.COLLECTION.TRANSACTION_${transactionId}`;
    }
}
```

Reply:
@tgolen 
 > OK thanks! Bear with me as I try to figure out how all this works. It seems to me like if there is a reportAction that you're connected to, and it has a reference to a transactionID, then when the transaction is created, the transactionID of the reportAction would go from null to <transactionID. At that point, you can connect to the transaction (via withOnyx) and have access to it.
Or am I still missing something?

That is what I expected. But from what i remember that did not happen. The `transcaction` seems to be created sometime after the `transactionID` of `reportAction` is generated

Reply:
@tgolen Feel free to try this approach if it makes sense 
https://github.com/Expensify/App/pull/25906#discussion_r1358588405

Reply:
What is the code for `getTransaction(lastReportAction);`?

Reply:
Something like this  https://github.com/Expensify/App/pull/25906/files/4811977a3acb278ca1e210601eb521584ba8db95#diff-020f2493b0feaf1b3762bed3d5d7ad6c28b5ef52b9e9c0a9fc8eb9673d270478
![image](https://github.com/Expensify/App/assets/65986357/039311ff-4505-4090-b89f-64a8c59090c4)

Reply:
> getTransaction(lastReportAction);

OK, so it's getting the transactionID like this `transactionID = reportAction.originalMessage?.IOUTransactionID ?? '';` which is fine.

> The transcaction seems to be created sometime after the transactionID of reportAction is generated. So you don't have access to transcation when the reportAction gets the transactionID

I don't think that matters. As long as you have the `transactionID`, you can connect to Onyx with it and you'll get any updates to the object on that key.

I imagine the code should look more like this:
```
receiptTransaction: {
    key: ({ reportActions }) => {
        const lastReportAction = _.last(reportActions);
        if (!lastReportAction || !lastReportAction.name || lastReportAction.name !== CONST.REPORT.ACTIONS.TYPE.IOU || !lastReportAction.originalMessage || !lastReportAction.originalMessage.IOUTransactionID) {
            return `${ONYXKEYS.COLLECTION.TRANSACTION}0`; // This will always return `null` for these cases
        }

        return `${ONYXKEYS.COLLECTION.TRANSACTION}${lastReportAction.originalMessage.IOUTransactionID}`;
    }
}
```

Reply:
@tgolen What you say makes sense. But I'm not sure how Onyx works in detail. From what i remember this approach didn't work for me. If it does for you, the change can be implemented

Reply:
You are right. Honestly, I'm very curious to know what is the actual issue here. 
I did not see it as a workaround, just the most I could do and explaining the reason for the same.